python/cudf/cudf/pylibcudf_tests/common/utils.py

```diff
@@ -0,0 +1,105 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+
+from typing import Optional
+
+import pyarrow as pa
+import pytest
+
+from cudf._lib import pylibcudf as plc
+
+
+def metadata_from_arrow_array(
+    pa_array: pa.Array,
+) -> Optional[plc.interop.ColumnMetadata]:
+    metadata = None
+    if pa.types.is_list(dtype := pa_array.type) or pa.types.is_struct(dtype):
+        metadata = plc.interop.ColumnMetadata(
+            "",
+            # libcudf does not store field names, so just match pyarrow's.
+            [
+                plc.interop.ColumnMetadata(pa_array.type.field(i).name)
+                for i in range(pa_array.type.num_fields)
+            ],
+        )
+    return metadata
+
+
+def assert_column_eq(plc_column: plc.Column, pa_array: pa.Array) -> None:
+    """Verify that the pylibcudf array and PyArrow array are equal."""
+    # Nested types require children metadata to be passed to the conversion function.
+    plc_pa = plc.interop.to_arrow(
+        plc_column, metadata=metadata_from_arrow_array(pa_array)
+    )
+
+    if isinstance(plc_pa, pa.ChunkedArray):
+        plc_pa = plc_pa.combine_chunks()
+    if isinstance(pa_array, pa.ChunkedArray):
+        pa_array = pa_array.combine_chunks()
+
+    assert plc_pa.equals(pa_array)
+
+
+def assert_table_eq(plc_table: plc.Table, pa_table: pa.Table) -> None:
+    """Verify that the pylibcudf array and PyArrow array are equal."""
+    plc_shape = (plc_table.num_rows(), plc_table.num_columns())
+    assert plc_shape == pa_table.shape
+
+    for plc_col, pa_col in zip(plc_table.columns(), pa_table.columns):
+        assert_column_eq(plc_col, pa_col)
+
+
+def cudf_raises(expected_exception: BaseException, *args, **kwargs):
+    # A simple wrapper around pytest.raises that defaults to looking for cudf exceptions
+    match = kwargs.get("match", None)
+    if match is None:
+        kwargs["match"] = "CUDF failure at"
+    return pytest.raises(expected_exception, *args, **kwargs)
+
+
+# TODO: Consider moving these type utilities into pylibcudf.types itself.
```

Comment:
We could also consider making them methods of `plc.DataType`

Reply:
Good point. Do you have a preference? And should we do that in a follow-up PR?

Reply:
Let's do it in a follow up. I'd prefer making them methods as pyarrow (and incidentally Polars) does that. But as Nick would say, it's a weakly held opinion